File: dsa_leetcode_beginner/0739-daily-temperatures/0739-daily-temperatures.cpp
#include <vector>
#include <stack>

class Solution {
public:
    std::vector<int> dailyTemperatures(std::vector<int>& temperatures) {
        int n = temperatures.size();
        std::vector<int> result(n, 0);
        std::stack<int> stack;

        for (int i = n - 1; i >= 0; i--) {
            while (!stack.empty() && temperatures[i] >= temperatures[stack.top()]) {
                stack.pop();
            }

            if (!stack.empty()) {
                result[i] = stack.top() - i;
            }

            stack.push(i);
        }

        return result;
    }
};

File: dsa_leetcode_beginner/0739-daily-temperatures/0739-daily-temperatures.js
function dailyTemperatures(temperatures) {
    const n = temperatures.length;
    const result = new Array(n).fill(0);
    const stack = [];

    for (let i = n - 1; i >= 0; i--) {
        while (stack.length > 0 && temperatures[i] >= temperatures[stack[stack.length - 1]]) {
            stack.pop();
        }

        if (stack.length > 0) {
            result[i] = stack[stack.length - 1] - i;
        }

        stack.push(i);
    }

    return result;
}
